server/src/main/scala/org/http4s/server/middleware/ResponseTiming.scala

```diff
@@ -0,0 +1,38 @@
+package org.http4s
+package server
+package middleware
+
+import cats.data.Kleisli
+import cats.effect._
+import cats.implicits._
+import org.http4s.util.CaseInsensitiveString
+
+import scala.concurrent.duration._
+
+object ResponseTiming {
+
+  /**
+    * Simple middleware for adding a custom header with timing information to a response.
+    *
+    * This middleware captures the time from when the request headers are parsed and supplied
+    * to the wrapped service to when the response is started. Metrics middleware, like this one,
+    * work best as the outer layer to ensure work done by other middleware is also included.
+    *
+    * @param http [[HttpApp]] to transform
+    * @param timeUnit the units of measure for this timing
+    * @param headerName the name to use for the header containing the timing info
+    */
+  def apply[F[_]: Sync](
```

Comment:
Consistency observation: Most of our middlewares are `apply(arg1: Blah, arg2: Foo)(http: HttpApp[F])`, so the `HttpApp` is in the final argument list.  Since most of them take implicit constraints, I also don't know how much that convention actually helps us in terms of partial application.

We also usually don't use context bounds when there are other implicits: I would move `Sync` into the list ahead of `clock`.

Reply:
If I change the arg list, then application looks a bit weird because of the default arguments: 
```scala
val app = ResponseTiming()(thisService)
```
So I tried to follow [EntityLimiter](https://github.com/http4s/http4s/blob/b3849cf830aa5a1bb51dd8d594e2d8edc45eaee9/server/src/main/scala/org/http4s/server/middleware/EntityLimiter.scala#L18) in this case. But I'm happy to change it to whichever is preferred.